enhancements/aws-ipv6-dev-env.md

```diff
@@ -0,0 +1,163 @@
+---
+title: AWS-IPv6-Dev-Env
+authors:
+  - "@russellb"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2019-10-24
+last-updated: 2019-10-24
+status: implementable
+---
+
+# aws-ipv6-dev-env
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+While Kubernetes itself includes support for IPv6, the OKD distribution does
+not fully support it yet.  To enable developers across the many components that
+make up OKD, this proposal is to add support for enabling IPv6 for an AWS
+cluster.  This will also be used to create a CI job that can exercise IPv6
+support across OKD.
+
+## Motivation
+
+Supporting IPv6 across OKD will require collaboration and fixes across many
+components.  Many developers are already used to doing development against AWS
+clusters, so supporting AWS as an IPv6 development environment should make it
+easier for people to jump in to help debug and fix IPv6 issues.
+
+### Goals
+
+* Make creating an IPv6 development environment as easy as creating an AWS
+  cluster, by just setting one additional option to turn it on.
+* Support OKD in reaching a milestone of supporting a single stack IPv6 control
+  plane (IPv6 only).
+
+### Non-Goals
+
+* Reaching full support for IPv6 on AWS.  While the work will get us closer,
+  that is not the primary goal here.  The primary goal is to help make it
+  easier for all teams to exercise IPv6.
+
+## Proposal
+
+To enable IPv6 in your AWS environment, a developer would set the following
+environment variable before running the installer:
+
+```bash
+    export OPENSHIFT_AWS_USE_IPV6=”true”
+```
+
+This would be off by default.
+
+### AWS Network Environment
+
+AWS does not support a single-stack (IPv6 only) environment, but it does
+support a dual-stack (IPv4 and IPv6) environment, so that’s what is enabled
+here.  This is a summary of the changes to the network environment:
+
+* The VPC has IPv6 enabled and a `/56` IPv6 CIDR will be allocated by AWS.
+* Each Subnet will have an IPv6 `/64` subnet allocated to it.
+* All IPv4 specific security group rules have corresponding IPv6 rules created.
+* AWS Network Load Balancers (NLBs) do not support IPv6, so external API access
+  is still over IPv4.  AWS does not have a TCP load balancer that supports
+  IPv6, other than classic load balancers with EC2-Classic, and not EC2-VPC.
+  AWS Application Load Balancers supposedly support IPv6, but that would
+  require doing HTTPS load balancing for the API instead of just TCP load
+  balancing, so we just use the IPv4 NLBs.  API access within the cluster is
+  still exercising IPv6.
+* IPv6 DNS records (AAAA) are created and the IPv4 (A) records are disabled,
```

Comment:
> IPv4 (A) records are disabled

what does this mean?

Reply:
It means we put IPv6 addresses in DNS instead of IPv4 where possible.  :-)

I'll expand on it further clarifying why the API records are still IPv4.